test/alignak_test.py

```diff
@@ -271,10 +239,95 @@ def fake_check(self, ref, exit_status, output="OK"):
+        self.schedulers[0].sched.waiting_results.put(check)
+
+    def scheduler_loop(self, count, items, reset_checks=True):
+        """
+        Manage scheduler checks
+
+        !!!!!!!!!! This function is to be replaced by the scheduler_loop_new !!!!!!!!!!
+
+
+        @verified
+        :param count: number of checks to pass
+        :type count: int
+        :param items: list of list [[object, exist_status, output]]
+        :type items: list
+        :return: None
+        """
+        if reset_checks:
+            self.schedulers[0].sched.checks = {}
+        for num in range(count):
+            for item in items:
+                (obj, exit_status, output) = item
+                obj.next_chk = time.time()
+                chk = obj.launch_check(obj.next_chk,
+                                       self.schedulers[0].sched.hosts,
+                                       self.schedulers[0].sched.services,
+                                       self.schedulers[0].sched.timeperiods,
+                                       self.schedulers[0].sched.macromodulations,
+                                       self.schedulers[0].sched.checkmodulations,
+                                       self.schedulers[0].sched.checks,
+                                       force=False)
+                self.schedulers[0].sched.add_check(chk)
+                # update the check to add the result
+                chk.set_type_active()
+                chk.output = output
+                chk.exit_status = exit_status
+                self.schedulers[0].sched.waiting_results.put(chk)
+            for i in self.schedulers[0].sched.recurrent_works:
+                (name, fun, nb_ticks) = self.schedulers[0].sched.recurrent_works[i]
+                if nb_ticks == 1:
+                    fun()
+
+    def scheduler_loop_new(self, count, items):
+        """
+        Manage scheduler checks
+
+        !!!!!!!!!! This function will replace the scheduler_loop !!!!!!!!!!
+
+        @verified
+
+        :param count: number of checks to pass
+        :type count: int
+        :param items: list of list [[object, exist_status, output]]
+        :type items: list
+        :return: None
+        """
+        for num in range(count):
+            for item in items:
+                (obj, exit_status, output) = item
+                if len(obj.checks_in_progress) == 0:
+                    for i in self.schedulers[0].sched.recurrent_works:
+                        (name, fun, nb_ticks) = self.schedulers[0].sched.recurrent_works[i]
+                        if nb_ticks == 1:
+                            fun()
+                self.assertGreater(len(obj.checks_in_progress), 0)
+                chk = self.schedulers[0].sched.checks[obj.checks_in_progress[0]]
+                chk.set_type_active()
+                chk.output = output
+                chk.exit_status = exit_status
+                self.schedulers[0].sched.waiting_results.put(chk)
```

Comment:
can you add a bit more comment? From what I get : we create new check by running recurrent work if necessary. Then you manully do the launch check part and make check to get consumed (recurrent works)